ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+(function() {
+
+
+    "use strict";
+    class CheckBoxGroup extends FormView.FormFieldBase {
+
+        static NS = FormView.Constants.NS;
+        /**
+         * Each FormField has a data attribute class that is prefixed along with the global namespace to
+         * distinguish between them. If a component wants to put a data-attribute X, the attribute in HTML would be
+         * data-{NS}-{IS}-x=""
+         * @type {string}
+         */
+        static IS = "adaptiveFormCheckBoxGroup";
+        static bemBlock = 'cmp-adaptiveform-checkboxgroup'
+        static selectors  = {
+            self: "[data-" + this.NS + '-is="' + this.IS + '"]',
+            widget: `.${CheckBoxGroup.bemBlock}__widget`,
+            widgetLabel: `.${CheckBoxGroup.bemBlock}__widget__label`,
+            label: `.${CheckBoxGroup.bemBlock}__label`,
+            description: `.${CheckBoxGroup.bemBlock}__longdescription`,
+            qm: `.${CheckBoxGroup.bemBlock}__questionmark`,
+            errorDiv: `.${CheckBoxGroup.bemBlock}__errormessage`
+        };
+
+        constructor(params) {
+            super(params);
+            this.qm = this.element.querySelector(CheckBoxGroup.selectors.qm)
+            this.widgetLabel = this.element.querySelector(CheckBoxGroup.selectors.widgetLabel)
+        }
+
+        getWidget() {
+            return this.element.querySelectorAll(CheckBoxGroup.selectors.widget);
+        }
+
+        getDescription() {
+            return this.element.querySelector(CheckBoxGroup.selectors.description);
+        }
+
+        getLabel() {
+            return this.element.querySelector(CheckBoxGroup.selectors.label);
+        }
+
+        getErrorDiv() {
+            return this.element.querySelector(CheckBoxGroup.selectors.errorDiv);
+        }
+
+        setModel(model) {
+            super.setModel(model);
+            let widgets = this.widget
+            widgets.forEach(widget => {
```

Comment:
does this work. I thought widgets is an HTMLCollection and doesn't have forEach method

Reply:
This works. Verified on runtime.